apitools/base/py/transfer.py

```diff
@@ -913,7 +918,22 @@ def CallSendMediaBody(start):
             if response.status_code in (http_client.OK, http_client.CREATED):
                 self.__complete = True
                 break
-            self.__progress = self.__GetLastByte(response.info['range'])
+            if response.status_code not in (
+                    http_client.OK, http_client.CREATED,
+                    http_wrapper.RESUME_INCOMPLETE):
+                # We want to reset our state to wherever the server left us
+                # before this failed request, and then raise.
+                self.RefreshResumableUploadState()
```

Comment:
Nit: Can you move the call to RefreshResumableUploadState() down past line 933, so that it only takes place after we know the strategy is "RESUMABLE_UPLOAD"?  I know RefreshResumableUploadState() will return if it's not a resumable upload, but it would look semantically cleaner if we only called it once we knew the upload was a resumable one :)

Reply:
Done!